internal/core/app_deploy.go

```diff
@@ -313,14 +315,48 @@ func (op *deployOperation) Do(ctx context.Context, log hclog.Logger, app *App, m
 		return nil, err
 	}
 
+	// NOTE(izaak): We to this to satisfy argmamper - because the Deploy function asks for this,
+	// we need to prove to argmapper that we can supply one. We'll swap this out
+	// with a different pointer on the plugin side. This one doesn't end up being used for anything.
+	declaredResourcesResp := &component.DeclaredResourcesResp{}
+
+	var args []argmapper.Arg
+	args = append(args, argmapper.Typed(declaredResourcesResp))
+	args = append(args, op.args()...)
+
 	val, err := app.callDynamicFunc(ctx,
 		log,
 		(*component.Deployment)(nil),
 		op.component,
 		op.component.Value.(component.Platform).DeployFunc(),
-		op.args()...,
+		args...,
```

Comment:
👍 good call. I had this up here initially because I was hoping I could get that declaredResourcesResp actually populated after calling the plugin func (like a real outparameter), but I had no such luck.

Reply:
I added this back - we need to hand onto a reference to the declaredResourceResp here in Do().